.github/workflows/consistency-functional.yml

```diff
@@ -122,9 +122,7 @@ jobs:
           sudo apt install python3-httplib2 -y
           # We need to be sure we use the latest versions of
           # pip, virtualenv and setuptools
-          # FIXME: `git revert` the pip version constraint when devstack is fixed for new pip
-          #        https://github.com/os-migrate/os-migrate/issues/317
-          sudo python -m pip install --upgrade "pip<20.3"
+          sudo python -m pip install --upgrade pip
```

Comment:
Have you thought about using a test-requirements.txt file or something to pin these and then use something like https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/keeping-your-dependencies-updated-automatically to keep them up to date via PR whenever they change. That way the updates are tested before they're used.

Reply:
We haven't, i think something like this would only make sense investing in if problems that it could prevent would happen regularly. Even in the case of this particular PR though, the solution you point out wouldn't help, because we needed to patch Devstack too.